plugins/transport/socket/main.go

```diff
@@ -80,17 +81,48 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	pc, err := net.ListenUnixgram("unixgram", &laddr)
 	if err != nil {
 		s.logger.Errorf(err, "failed to bind unix socket %s", laddr.Name)
-		return
+		return nil
 	}
 	// create socket file if it does not exist
 	skt, err := pc.File()
 	if err != nil {
 		s.logger.Errorf(err, "failed to retrieve file handle for %s", laddr.Name)
-		return
+		return nil
 	}
 	skt.Close()
 
 	s.logger.Infof("socket listening on %s", laddr.Name)
+
+	return pc
+}
+
+func (s *Socket) initUdpSocket() *net.UDPConn {
+	addr, err := net.ResolveUDPAddr("udp", s.conf.Url)
```

Comment:
If you don't want to explicitly test `err` (since I assume errors in resolution would be caught on the next line in the ListenUDP call) then you should do `addr, _ := ...`. Handling the error explicitly could result in a better troubleshooting experience, though.

Reply:
Yup having information in log (at least) about issue with socket address would improve debugging.

Reply:
Good catch, I overlooked that one, I'll add a test for the error there.